src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -659,5 +660,32 @@ public override object Clone()
         {
             return new ConsensusManagerBehavior(this.chain, this.initialBlockDownloadState, this.consensusManager, this.peerBanning, this.loggerFactory);
         }
+
+        internal int GetCachedItemsCount()
+        {
+            return this.cachedHeaders.Count;
+        }
+
+        /// <summary>
+        /// Gets the count of consumed headers in a <paramref name="headers"/> list, giving a <paramref name="consumedHeader"/> reference.
+        /// All items up to <paramref name="consumedHeader"/> are considered consumed.
+        /// </summary>
+        /// <param name="headers">List of headers to use to get the consumed count.</param>
+        /// <param name="consumedHeader">The consumed header reference.</param>
+        /// <returns>The number of consumed cached items.</returns>
+        private int GetConsumedHeadersCountLockedAsync(List<BlockHeader> headers, BlockHeader consumedHeader)
```

Comment:
its not async

Reply:
when there is locked in the name there should be a comment referencing by which lock it should be locked

And I don't think it should be locked

Reply:
both method calls are under lock, so I added the Lock to the name even because one of the call need cachedItems that must be under lock (initially the method didn't had the headers argument but was using only cachedHeaders but Dan asked to be used on both places, so I had to add an headers argument to be able to use it in the part of code that doesn't use cachedHeaders.

I'm not sure what to do in this case, maybe It's ok to remove the Locked at this point

Reply:
In one case of this method call we use `this.cachedHeaders` which should be called under a lock and in both cases we are anyway the `asyncLock` so I'll remove the `async` and keep the `lock`.

Reply:
Actually I will remove the `Locked`  this is up to the caller and does not make sense in this context